pulsar/producer_partition.go

```diff
@@ -827,6 +998,10 @@ func (p *partitionProducer) Send(ctx context.Context, msg *ProducerMessage) (Mes
 
 	// wait for send request to finish
 	<-doneCh
+
+	// handle internal error
+	p.internalErrHandle(err)
```

Comment:
Why do we need to introduce the internal error handler here? Can we handle this error inside the sending operation?

Reply:
If we handle this error inside the sending operation, we can only do it in following.
https://github.com/apache/pulsar-client-go/blob/2d5f6fcfa22af4184793a22ca353731a02b54ca9/pulsar/producer_partition.go#L627-L636
I don't think it's obvious, do you have a better idea?